From 0591ce4ade09152ec3bcb5b416475b57d828443d Mon Sep 17 00:00:00 2001
From: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>
Date: Mon, 21 Jun 2021 18:42:00 +0300
Subject: [PATCH] r8a7795(6|65).dtsi: Add multichannel audio ranges that
 AUDMACs relies on

The 1:1 translations based on these ranges should reach IPMMU pagetables
for AUDMACs to work through it.
The one possible way to do it is to describe them in the device tree.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Signed-off-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>
---
 arch/arm64/boot/dts/renesas/r8a77951.dtsi | 3 ++-
 arch/arm64/boot/dts/renesas/r8a77961.dtsi | 3 ++-
 arch/arm64/boot/dts/renesas/r8a77965.dtsi | 3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77951.dtsi b/arch/arm64/boot/dts/renesas/r8a77951.dtsi
index 9989c1c..2ef808a 100644
--- a/arch/arm64/boot/dts/renesas/r8a77951.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77951.dtsi
@@ -2436,7 +2436,8 @@
 				<0 0xec541000 0 0x280>,  /* SSI */
 				<0 0xec760000 0 0x200>,  /* Audio DMAC peri peri*/
 				<0 0xec000000 0 0x1000>,
-				<0 0xec008000 0 0x1000>;
+				<0 0xec008000 0 0x1000>,
+				<0 0xec100000 0 0x100000>;
 			reg-names = "scu", "adg", "ssiu", "ssi", "audmapp";
 
 			clocks = <&cpg CPG_MOD 1005>,
diff --git a/arch/arm64/boot/dts/renesas/r8a77961.dtsi b/arch/arm64/boot/dts/renesas/r8a77961.dtsi
index cae5822..6642e41 100644
--- a/arch/arm64/boot/dts/renesas/r8a77961.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77961.dtsi
@@ -2141,7 +2141,8 @@
 			      <0 0xec541000 0 0x280>,  /* SSI */
 			      <0 0xec760000 0 0x200>,  /* Audio DMAC peri peri*/
 			      <0 0xec000000 0 0x1000>,
-			      <0 0xec008000 0 0x1000>;
+			      <0 0xec008000 0 0x1000>,
+			      <0 0xec100000 0 0x100000>;
 			reg-names = "scu", "adg", "ssiu", "ssi", "audmapp";
 
 			clocks = <&cpg CPG_MOD 1005>,
diff --git a/arch/arm64/boot/dts/renesas/r8a77965.dtsi b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
index 19d371e..4cd542a 100644
--- a/arch/arm64/boot/dts/renesas/r8a77965.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77965.dtsi
@@ -1636,7 +1636,8 @@
 				<0 0xec541000 0 0x280>,  /* SSI */
 				<0 0xec760000 0 0x200>,  /* Audio DMAC peri peri*/
 				<0 0xec000000 0 0x1000>,
-				<0 0xec008000 0 0x1000>;
+				<0 0xec008000 0 0x1000>,
+				<0 0xec100000 0 0x100000>;
 			reg-names = "scu", "adg", "ssiu", "ssi", "audmapp";
 
 			clocks = <&cpg CPG_MOD 1005>,
-- 
2.7.4

